runtime/query_test.go

```diff
@@ -338,6 +347,51 @@ func TestPopulateParameters(t *testing.T) {
 	}
 }
 
+func TestPopulateParametersWithNativeTypes(t *testing.T) {
+	timeT := time.Date(2016, time.December, 15, 12, 23, 32, 49, time.UTC)
+	timeStr := timeT.Format(time.RFC3339Nano)
+
+	durationT := 13 * time.Hour
+	durationStr := durationT.String()
+
+	for _, spec := range []struct {
+		values url.Values
+		want   *nativeProto3Message
+	}{
+		{
+			values: url.Values{
+				"native_timestamp_value": {timeStr},
+				"native_duration_value":  {durationStr},
+			},
+			want: &nativeProto3Message{
+				NativeTimeValue:     &timeT,
+				NativeDurationValue: &durationT,
+			},
+		},
+		{
+			values: url.Values{
+				"nativeTimestampValue": {timeStr},
+				"nativeDurationValue":  {durationStr},
+			},
+			want: &nativeProto3Message{
+				NativeTimeValue:     &timeT,
+				NativeDurationValue: &durationT,
+			},
+		},
+	} {
+		msg := new(nativeProto3Message)
+		err := runtime.PopulateQueryParameters(msg, spec.values, utilities.NewDoubleArray(nil))
+
+		if err != nil {
+			t.Errorf("runtime.PopulateQueryParameters(msg, %v, utilities.NewDoubleArray(nil)) failed with %v; want success", spec.values, err)
+			continue
+		}
+		if got, want := msg, spec.want; !proto.Equal(got, want) {
+			t.Errorf("runtime.PopulateQueryParameters(msg, %v, utilities.NewDoubleArray(nil)) = %v; want %v", spec.values, got, want)
+		}
+	}
+}
```

Comment:
This needs to be a separate test because `golang/protobuf/proto.Clone` panics when using *time.Time in the proto message.